workflow.rst

```diff
@@ -452,6 +444,155 @@ missing a title::
 
     The optional second argument of ``setBlocked()`` was introduced in Symfony 5.1.
 
+.. _workflow-usage-choose-events:
+
+Choosing which Events to dispatch
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 5.2
+
+    Ability to choose which events to dispatch was introduced in Symfony 5.2.
+
+You are able to specify which events (does not apply to Guard event) will be
+fired when performing each transition by passing an array of workflow events
+to the ``events_to_dispatch`` configuration option.
+
+Valid options for ``events_to_dispatch`` are:
+
+    * ``null`` - all events are dispatched
+    * ``[]`` - no events are dispatched (except the Guard event)
+    * ``['workflow.leave', 'workflow.completed']`` - specific events are dispatched
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/workflow.yaml
+        framework:
+            workflows:
+                blog_publishing:
+                    # ...
+                    events_to_dispatch: ['workflow.leave', 'workflow.completed']
+                    # ...
+
+    .. code-block:: xml
+
+        <!-- config/packages/workflow.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony https://symfony.com/schema/dic/symfony/symfony-1.0.xsd"
+        >
+            <framework:config>
+                <framework:workflow name="blog_publishing">
+                    <!-- ... -->
+                    <framework:event-to-dispatch>workflow.leave</framework:event-to-dispatch>
+                    <framework:event-to-dispatch>workflow.completed</framework:event-to-dispatch>
+                    <!-- ... -->
+                </framework:workflow>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/workflow.php
+        $container->loadFromExtension('framework', [
+            // ...
+            'workflows' => [
+                'blog_publishing' => [
+                    // ...
+                    'events_to_dispatch' => [
+                        'workflow.leave',
+                        'workflow.completed',
+                    ],
+                    // ...
+                ],
+            ],
+        ]);
+
+To specify that no events will be dispatched pass an empty array to the
+configuration option. Guard Events are still dispatched in this instance.
```

Comment:
```suggestion
configuration option.

.. note::

    Guard Events are still dispatched in this instance.
```

What do you think of making this sentence more prominent?

Reply:
also i would put this note on top
after the `Valid options for ``events_to_dispatch`` are:`
as it is documented ` (except the Guard event)`

Reply:
I've added the note further up as @noniagriconomie suggested, does that look alright?